tests/contrib/hooks/test_bigquery_hook.py

```diff
@@ -729,5 +702,21 @@ def test_legacy_sql_override_propagates_properly(self, run_with_config):
             self.assertIs(args[0]['query']['useLegacySql'], False)
 
 
+class TestBigQueryHookLocation(unittest.TestCase):
+    @mock.patch.object(hook.BigQueryBaseCursor, 'run_with_configuration')
+    def test_location_propagates_properly(self, run_with_config):
+        with mock.patch.object(hook.BigQueryHook, 'get_service'):
+            bq_hook = hook.BigQueryHook(location=None)
+            self.assertIsNone(bq_hook.location)
+
+            bq_cursor = hook.BigQueryBaseCursor(mock.Mock(),
+                                                'test-project',
+                                                location=None)
+            self.assertIsNone(bq_cursor.location)
+            bq_cursor.run_query(sql='select 1', location='US')
```

Comment:
Why do you need to specify `location` on `run_query` after initialization of `BigQueryHook`?

Reply:
It gives an option to override defaults